tests/chainer_tests/training_tests/extensions_tests/test_logreport.py

```diff
@@ -0,0 +1,64 @@
+import json
+import os.path
+import shutil
+import tempfile
+import unittest
+
+import mock
+
+from chainer import testing
+from chainer import training
+from chainer.training import extensions
+
+
+class TestLogReport(unittest.TestCase):
+
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_dir)
+
+    def test_trigger(self):
+        trainer = _get_mocked_trainer((10, 'iteration'), self.temp_dir)
+        log_report = extensions.LogReport(trigger=(1, 'iteration'))
+        trainer.extend(log_report)
+        trainer.run()
+        with open(os.path.join(self.temp_dir, 'log')) as f:
+            self.assertEqual(log_report.log, json.load(f))
+
+    def test_write_trigger(self):
+        stops, aggregates, writes = 10, 2, 5
+        trainer = _get_mocked_trainer((stops, 'iteration'), self.temp_dir)
+        with mock.patch.object(extensions.LogReport, '_write') as mocked:
+            log_report = extensions.LogReport(
+                trigger=(aggregates, 'iteration'),
+                write_trigger=(writes, 'iteration')
+            )
+            trainer.extend(log_report)
+            trainer.run()
+            mocked.assert_called_with(log_report.log[-1], self.temp_dir)
+            self.assertEqual(mocked.call_count, stops // writes)
+            self.assertEqual(len(log_report.log), stops // aggregates)
+
+
+def _get_mocked_trainer(stop_trigger=(10, 'iteration'), out='result'):
```

Comment:
Chainer offers [chainer.testing.get_trainer_with_mock_updater](https://github.com/chainer/chainer/blob/master/chainer/testing/training.py#L8) from v2 for testing trainers and related modules. Could you use that instead of the custom one if possible?